From 182e24f22bb0d81eacc8d0397c8b6617ad673d31 Mon Sep 17 00:00:00 2001
From: Ron Yorston <rmy@pobox.com>
Date: Fri, 25 Aug 2023 07:31:39 +0100
Subject: [PATCH] Fix .DEFAULT rule for prerequisites

Commit 9b2292b (Fixes to inference rules) set the N_TARGET flag
for prerequisites being built recursively.  This wasn't necessary
for the issue at hand and it lead to other problems (busybox-w32
GitHub issue #354).

Undo that part of the commit.
---
 make.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/make.c b/make.c
index 06a0bf3..8bbc55f 100644
--- a/make.c
+++ b/make.c
@@ -285,7 +285,6 @@ make(struct name *np, int level)
 #endif
 		for (dp = rp->r_dep; dp; dp = dp->d_next) {
 			// Make prerequisite
-			dp->d_name->n_flag |= N_TARGET;
 			estat |= make(dp->d_name, level + 1);
 
 			// Make strings of out-of-date prerequisites (for $?),
